apps/api/src/app/environments/environments.bridge.controller.ts

```diff
@@ -0,0 +1,133 @@
+     *
+     * I'm leaving it here as a reminder that we need to remove the UI health-check for Novu-managed
+     * bridge environments.
+     */
+    const novuBridgeHandler = new NovuNestjsHandler({
+      workflows: [],
+      client: new Client({
+        strictAuthentication: false,
+        secretKey: await this.getApiKey(environmentId),
+      }),
+    });
+    await novuBridgeHandler.handleRequest(req, res, 'GET');
+  }
+
+  @Post('/:environmentId/bridge')
+  async postHandler(
+    @Req() req: Request,
+    @Res() res: Response,
+    @Param('environmentId') environmentId: string,
+    @Query('workflowId') workflowId: string,
+    @Body() event: Event
+  ) {
+    const foundWorkflow = await this.getWorkflow(environmentId, workflowId);
+
+    const programmaticallyCreatedWorkflow = this.createWorkflow(foundWorkflow, event.controls);
+
+    const novuBridgeHandler = new NovuNestjsHandler({
+      workflows: [programmaticallyCreatedWorkflow],
+      client: new Client({ strictAuthentication: true, secretKey: await this.getApiKey(environmentId) }),
+    });
+
+    await novuBridgeHandler.handleRequest(req, res, 'POST');
+  }
+
+  // TODO: wrap the secret key fetching per environmentId in a usecase, including the decryption, add caching.
+  private async getApiKey(environmentId: string): Promise<string> {
+    const environment = await this.environmentsRepository.findOne({ _id: environmentId });
+    if (!environment) {
+      throw new NotFoundException('Environment not found');
+    }
+    const secretKey = decryptApiKey(environment.apiKeys[0].key);
+
+    return secretKey;
+  }
+
+  // TODO: refactor this into a usecase, add caching.
+  private async getWorkflow(environmentId: string, workflowId: string): Promise<NotificationTemplateEntity> {
+    const foundWorkflow = await this.workflowsRepository.findByTriggerIdentifier(environmentId, workflowId);
+
+    if (!foundWorkflow) {
+      throw new NotFoundException('Workflow not found');
+    }
+
+    return foundWorkflow;
+  }
+
+  private createWorkflow(newWorkflow: NotificationTemplateEntity, controls: Record<string, unknown>): Workflow {
+    return workflow(
+      newWorkflow.name,
+      async ({ step }) => {
+        for await (const staticStep of newWorkflow.steps) {
+          await step[stepFnFromStepType[staticStep.template!.type]](staticStep.stepId, () => controls, {
+            // TODO: fix the step typings, `controls` lives on template property, not step
+            controlSchema: (staticStep.template as unknown as typeof staticStep).controls?.schema,
+            /*
+             * TODO: add conditions
+             * Used to construct conditions defined with https://react-querybuilder.js.org/ or similar
+             * skip: () => !foundWorkflow.preferences.channels[staticStep.type],
+             */
+          });
+        }
+      },
+      {
+        /*
+         * TODO: these are probably not needed, given that this endpoint focuses on execution only.
```

Comment:
That's right about tags, regarding preferences we should take it offline as we need persistence of the default values.

Reply:
Default value persistence is now handled in the CreateWorkflowV2 use-case.